### DISCUSSION TITLE
Pode.Web and Azure Functions

### DISCUSSION THREAD

**Discussion by JustinGrote** [Badgerati/Pode#1079] [General]
Hello,

I want to make an ARM template generator for Azure Lighthouse. Pode and Azure functions seems like a nice way to do it.

I'm trying to figure out the correct syntax to get an azure function to respond with a Pode Web Page. My goal eventually is to present a form, and when clicked, will submit to another azure function which will return a JSON result to be displayed in a box.

Using the base example, I'm trying to do something like this, but I can't figure out how to wire up the route to the webpage since I'm not defining an endpoint?

My HTTP Trigger

```powershell

# Input bindings are passed in via param block.
param($Request, $TriggerMetadata)

Start-PodeServer -Request $TriggerMetadata -ServerlessType AzureFunctions {
    Add-PodeRoute 
    Use-PodeWebTemplates -Title 'Example' -Theme Dark

    Add-PodeWebPage -Name 'Services' -Icon 'Settings' -ScriptBlock {
        New-PodeWebCard -Content @(
            New-PodeWebTable -Name 'Services' -ScriptBlock {
                foreach ($svc in (Get-Service)) {
                    [ordered]@{
                        Name   = $svc.Name
                        Status = "$($svc.Status)"
                    }
                }
            }
        )
    }
}
```

**Comment by Badgerati** [comment]
Hi @JustinGrote,

Pode.Web doesn't work in Azure Functions 😃 

Using the main Pode module you can create very simplistic [sites that work in Azure Functions](https://badgerati.github.io/Pode/Hosting/AzureFunctions/#websites); but due to the nature of how Azure Functions work, and how Pode.Web works, the two aren't compatible.

If you want to use Pode.Web your best bet would be Azure Web Apps or Azure Container Apps.

**Reply by JustinGrote** [reply]
Makes sense! I ended up just building a static site for a proof of concept, then decided I wanted it to have a shared codebase that could work fully client side but also allow for a generated link to the ARM template for the `Deploy to Azure` template, so I ended up just going Typescript with CloudFlare Workers/Pages. Thanks though! Was a fun experiment.
